gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -153,12 +165,51 @@ private ChannelAndRequestCounter buildManagedChannel() {
     return new ChannelAndRequestCounter(channel, counter);
   }
 
-  public StorageBlockingStub getBlockingStub() {
-    return StorageGrpc.newBlockingStub(getManagedChannel());
+  public StorageBlockingStub getBlockingStub() throws IOException {
+    StorageBlockingStub blockingStub = StorageGrpc.newBlockingStub(getManagedChannel());
+    try {
+      if (credential instanceof GoogleCredential) {
```

Comment:
In what cases it will be not `GoogleCredential`?

Reply:
When authenticating with default GCE vm service account, in that case we can assume it's running on GCE env and credential will be auto-loaded without the need to explicitly pass it in.

Reply:
In this case I think that it's better to be explicit and check that metadata server authentication is configured instead of checking credential class.

Reply:
Done.